doc/kb_sdk_refdata.md

```diff
@@ -0,0 +1,55 @@
+# Adding Reference Data to a KBase App
+
+## Motivation
+
+Some tools and applications make use of large reference data collections.  For
+example, a QC app may have a reference data with common contaminants.  Adding
+data such as this to the Git repo for the module can make it very large.
+In addition, the generated Docker image would also be very large.  KBase App
+reference data is designed to address this scenario.
+
+## How Reference Data works
+
+To minimize the image size, this feature works by saving reference data in a
+special volume space and then making that volume available through a read-only
+mount in /data at app execution time.  An init mode for the module is run at registration
+time to initialize the reference data volume.  In addition, the reference data
+is separately versioned, so code-only updates to a module can be made without
+having to re-initialize the reference data.  
+
+## Usage
+
+To make use of this feature, the app developer needs to do a few simple things.
+
+1. Add the data-version tag to the kbase.yaml file and give it a semantic version
+tag.
+2. Add any downloading and data preparation steps to the init section of the
+entrypoint.sh script in the scripts directory.  The data should be placed in
+/data and any apps that use the dated should be configured to look for the data
+in this location.
+3. Create a \_\_READY\_\_ file in /data to indicate that the reference data volume
+was successfully created.  Ideally, some sanity tests should be performed to confirm
+things ran correctly by testing for the presence of expected files.  If the
+\_\_READY\_\_ file is not present, the registration will fail and the reference data
+area will be removed.
+
+You can see an example in the RAST application:
+https://github.com/kbaseapps/RAST_SDK/blob/master/scripts/entrypoint.sh#L18-L28
```

Comment:
This should link to a specific commit as master may change and then the line highlighting may highlight the wrong stuff